python/CMakeLists.txt

```diff
@@ -56,3 +56,101 @@ endfunction(create_symlinks)
 
 file(GLOB_RECURSE python_files RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} *.py)
 create_symlinks(${python_files})
+
+################################################################################
+# Native extensions
+################################################################################
+
+include(NpcompPython)
+
+# Normally on unix-like platforms, extensions are built as "MODULE" libraries
+# and do not explicitly link to the python shared object. This allows for
+# come greater deployment flexibility since the extension will bind to
+# symbols in the python interpreter on load. However, it also keeps the
+# linker from erroring on undefined symbols, leaving this to (usually obtuse)
+# runtime errors. Building in "SHARED" mode with an explicit link to the
+# python libraries allows us to build with the expectation of no undefined
+# symbols, which is better for development.
+# TODO(laurenzo): Windows requires linking against the PYTHON_LIBRARIES
+# TODO(laurenzo): OSX requires allowing undefined (-undefined dynamic_lookup)
+set(NPCOMP_PYEXT_LINK_MODE SHARED)
+set(NPCOMP_PYEXT_LIBADD ${PYTHON_LIBRARIES})
```

Comment:
the meaning of "LIBADD" is not clear to me. "additional libraries"?

Reply:
This is just copied verbatim from another file, so I'm leaving it for now. Fwiw, this is autoconf nomenclature. Shortly, I'll be reworking this upstream into cmake helper functions and using that here. Will work out verbiage then.